sonar-text-dotnet/pipeline_scripts/build_and_pack_solution.ps1

```diff
@@ -6,19 +6,24 @@ function CheckIfSuccessful($StepName) {
 }
 
 Set-Location $env:PROJECT_DIR
-$SIGN_DOTNET_ASSEMBLY = "$CIRRUS_BRANCH" -eq "master" -or "$CIRRUS_BRANCH".startsWith("branch-")
-Write-Host "Should the assembly be signed: ${SIGN_DOTNET_ASSEMBLY}"
-
-Write-Host Restoring $env:SOLUTION_DIR
-$nugetFeed = "https://pkgs.dev.azure.com/sonarsource/399fb241-ecc7-4802-8697-dcdd01fbb832/_packaging/slvs_input/nuget/v3/index.json"
-# For nuget version see: https://github.com/SonarSource/re-ci-images/blob/master/ec2-images/build-base-windows-dotnet.pkr.hcl#L39
-nuget restore -LockedMode -Source $nugetFeed
-CheckIfSuccessful "nuget restore"
-
-Write-Host Building $env:SOLUTION_DIR
-dotnet build `
-    $env:SOLUTION_DIR `
-    --no-restore `
+
+# In order to allow trackable builds in artifactory, we need to supply the `build-name` and `build-number` to nuget and dotnet commands.
+$buildId = $env:BUILD_NUMBER
+$buildName = $env:ARTIFACTORY_BUILD_NAME
+$nugetFeed = "https://repox.jfrog.io/artifactory/api/nuget/nuget"
+
+Write-Host "Setting up artifactory authentication for QA deploy"
+jf config add "repox" --url https://repox.jfrog.io --access-token $env:ARTIFACTORY_DEPLOY_PASSWORD
+jf rt ping # Check if the artifactory configuration is successfull
+
+Write-Host "Setting up artifactory dotnet configuration"
+jf dotnet-config --global --server-id-resolve repox --repo-resolve $nugetFeed
+
+$signAssembly = "$env:CIRRUS_BRANCH" -eq "master" -or "$env:CIRRUS_BRANCH".startsWith("branch-")
+Write-Host "Should the assembly be signed: ${signAssembly}"
+
+Write-Host "Building project"
+jf dotnet build `
```

Comment:
For the rest of `dotnet` commands, excepting signing, we will use the `jf`` wrapper. This collects build information that will later be published to artifactory.